docs/admin/federation/index.md

```diff
@@ -0,0 +1,111 @@
+---
+---
+This guide explains how to set up a federation cluster. Federation is an alpha feature which lets you control multiple kubernetes clusters.
+[Federation proposal](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/federation.md) has more details.
+
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running kubernetes cluster.
+If not, then you can head over to the [getting started guides](/docs/getting-started-guides/) to bring up a cluster.
+
+This guide also assumes that you have the kubernetes source code.
+If not, then you can head over to the [download kubernetes guide](/docs/getting-started-guides/binary_release/) to get the source code.
+
+## Setting up federation control plane
+
+### Building and pushing federation control binaries
+
+You can build the federation control plane binaries using the following command:
+
+```shell
+$ FEDERATION=true KUBE_RELEASE_RUN_TESTS=n make quick-release
+```
+
+Next, you need to push the images to a registry from where your kubernetes cluster can read them.
+If your kubernetes cluster is running on GCE, then you can push the images to `gcr.io/<your-gce-project-name>`.
+The command to push the images will look like:
+
+```shell
+$ FEDERATION=true FEDERATION_PUSH_REPO_BASE=gcr.io/<your-gce-project-name> ./build/push-federation-images.sh
+```
+
+### Running the federation control plane
+
+Federation control plane consists of an etcd, federation-apiserver and federation-controller-manager.
+You can run these binaries as pods on your existing kubernetes cluster.
+The command to run these pods on an existing GCE cluster will look like:
+
+```shell
+$ KUBERNETES_PROVIDER=gce FEDERATION_PUSH_REPO_BASE=gcr.io/<your-gce-project-name> ./federation/cluster/federation-up.sh
+```
+
+This creates a namespace `federation` and creates 2 deployments: `federation-apiserver` and `federation-controller-manager`.
+You can verify that the pods are available by running the following command:
+
+```shell
+$ kubectl get deployments --namespace=federation
+NAME                            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+federation-apiserver            1         1         1            1           10m
+federation-controller-manager   1         1         1            1           10m
+```
+
+Running `federation-up.sh` also creates a new record in your kubeconfig for you
+to be able to talk to federation apiserver. You can view this by running
+`kubectl config view`.
+
+## Registering kubernetes clusters for federation
```

Comment:
Blank line btx heading and text plz

This goes for the entire PR, just mentioning it once here